mlos_bench/mlos_bench/storage/sql_schema.py

```diff
@@ -0,0 +1,170 @@
+            Column("status", String(16), nullable=False),
+
+            PrimaryKeyConstraint("exp_id", "trial_id"),
+            ForeignKeyConstraint(["exp_id"], [self.experiment.c.exp_id]),
+            ForeignKeyConstraint(["config_id"], [self.config.c.config_id]),
+        )
+
+        # Values of the tunable parameters of the experiment,
+        # fixed for a particular trial config.
+        self.config_param = Table(
+            "config_param",
+            self._meta,
+            Column("config_id", Integer, nullable=False),
+            Column("param_id", String(255), nullable=False),
+            Column("param_value", String(255)),
+
+            PrimaryKeyConstraint("config_id", "param_id"),
+            ForeignKeyConstraint(["config_id"], [self.config.c.config_id]),
+        )
+
+        # Values of additional non-tunable parameters of the trial,
+        # e.g., scheduled execution time, VM name / location, number of repeats, etc.
+        self.trial_param = Table(
+            "trial_param",
+            self._meta,
+            Column("exp_id", String(255), nullable=False),
+            Column("trial_id", Integer, nullable=False),
+            Column("param_id", String(255), nullable=False),
+            Column("param_value", String(255)),
+
+            PrimaryKeyConstraint("exp_id", "trial_id", "param_id"),
+            ForeignKeyConstraint(["exp_id", "trial_id"],
+                                 [self.trial.c.exp_id, self.trial.c.trial_id]),
+        )
+
+        self.trial_result = Table(
+            "trial_result",
+            self._meta,
+            Column("exp_id", String(255), nullable=False),
+            Column("trial_id", Integer, nullable=False),
+            Column("metric_id", String(255), nullable=False),
+            Column("metric_value", String(255)),
+
+            PrimaryKeyConstraint("exp_id", "trial_id", "metric_id"),
+            ForeignKeyConstraint(["exp_id", "trial_id"],
+                                 [self.trial.c.exp_id, self.trial.c.trial_id]),
+        )
+
+        self.trial_telemetry = Table(
+            "trial_telemetry",
+            self._meta,
+            Column("exp_id", String(255), nullable=False),
+            Column("trial_id", Integer, nullable=False),
+            Column("ts", DateTime, nullable=False, default="now"),
+            Column("metric_id", String(255), nullable=False),
+            Column("metric_value", String(255)),
+
+            UniqueConstraint("exp_id", "trial_id", "ts", "metric_id"),
+            ForeignKeyConstraint(["exp_id", "trial_id"],
+                                 [self.trial.c.exp_id, self.trial.c.trial_id]),
+        )
+
+        _LOG.debug("Schema: %s", self._meta)
+
+    def create(self) -> 'DbSchema':
+        """
+        Create the DB schema.
+        """
+        _LOG.info("Create the DB schema")
+        self._meta.create_all(self._engine)
+        return self
+
+    def __repr__(self) -> str:
```

Comment:
Maybe a small comment here about how it can be used to dump the schema to text?

Reply:
done!